library/src/blas3/Tensile/gemm.cpp

```diff
@@ -54,83 +26,510 @@ const char* tensileGetSolutionName(rocblas_operation trans_a,
+                  "internal rocblas_float_complex does not match public rocblas_float_complex");
+
+    hipError_t status = hipErrorInvalidValue;
+
+    switch(GetTransposeMode(trans_a, trans_b))
+    {
+    case NN:
+        status = tensile_Cijk_Ailk_Bljk_CB(TENSILE_ARGS(TensileComplexFloat));
+        break;
+    case NT:
+        status = tensile_Cijk_Ailk_Bjlk_CB(TENSILE_ARGS(TensileComplexFloat));
+        break;
+    case TN:
+        status = tensile_Cijk_Alik_Bljk_CB(TENSILE_ARGS(TensileComplexFloat));
+        break;
+    case TT:
+        status = tensile_Cijk_Alik_Bjlk_CB(TENSILE_ARGS(TensileComplexFloat));
+        break;
+    case NC:
+        status = tensile_Cijk_Ailk_BjlkC_CB(TENSILE_ARGS(TensileComplexFloat));
+        break;
+    case CN:
+        status = tensile_Cijk_AlikC_Bljk_CB(TENSILE_ARGS(TensileComplexFloat));
+        break;
+    case TC:
+        status = tensile_Cijk_Alik_BjlkC_CB(TENSILE_ARGS(TensileComplexFloat));
+        break;
+    case CT:
+        status = tensile_Cijk_AlikC_Bjlk_CB(TENSILE_ARGS(TensileComplexFloat));
+        break;
+    case CC:
+        status = tensile_Cijk_AlikC_BjlkC_CB(TENSILE_ARGS(TensileComplexFloat));
+        break;
+    }
+
+    return status;
+}
+
+template <>
+hipError_t tensile_helper(rocblas_double_complex&       alpha_h,
+                          rocblas_double_complex&       beta_h,
+                          const rocblas_double_complex* A,
+                          const rocblas_double_complex* B,
+                          rocblas_double_complex*       C,
+                          rocblas_operation             trans_a,
+                          rocblas_operation             trans_b,
+                          rocblas_int                   strideC1,
+                          rocblas_int                   strideC2,
+                          rocblas_int                   strideA1,
+                          rocblas_int                   strideA2,
+                          rocblas_int                   strideB1,
+                          rocblas_int                   strideB2,
+                          rocblas_int                   sizeI,
+                          rocblas_int                   sizeJ,
+                          rocblas_int                   sizeK,
+                          rocblas_int                   sizeL,
+                          rocblas_handle                handle)
+{
+    static_assert(std::is_standard_layout<TensileComplexDouble>{},
+                  "rocblas_float_complex is not a standard layout type, and thus is "
+                  "incompatible with C.");
+
+    static_assert(std::is_trivial<TensileComplexDouble>{},
+                  "rocblas_float_complex is not a trivial type, and thus is "
+                  "incompatible with C.");
+
+    static_assert(sizeof(rocblas_double_complex) == sizeof(TensileComplexDouble),
+                  "internal rocblas_float_complex does not match public rocblas_double_complex");
```

Comment:
The comment is a little off.

Reply:
understand. update it in commit "update complex change 2".